Makefile.include: `clean-intermediate` deletes `esp32_bootloader` subfolder in `bin/`

#### Description

As the title suggests, the `clean-intermediate` target used by `compile_and_test_for_board.py` when using the `--clean-after` command deletes all subdirectories of the `bin/` folder of an application or example.

While this is good to clean up intermediate build products, it also cleans up the `esp32_bootloader` subfolder which makes it impossible to flash an ESP32 😅 

#### Steps to reproduce the issue and actual results

Either you can run the script, for example with `examples/advanced/psa_crypto`
```
./dist/tools/compile_and_test_for_board/compile_and_test_for_board.py --applications examples/advanced/psa_crypto --clean-after --incremental --jobs 4 . esp32-wemos-d1-r32

INFO:esp32-wemos-d1-r32.examples/advanced/psa_crypto:Board supported: True
INFO:esp32-wemos-d1-r32.examples/advanced/psa_crypto:Board has enough memory: True
INFO:esp32-wemos-d1-r32.examples/advanced/psa_crypto:Application has test: True
INFO:esp32-wemos-d1-r32.examples/advanced/psa_crypto:Run compilation
INFO:esp32-wemos-d1-r32.examples/advanced/psa_crypto:Nothing to be done for compilation
INFO:esp32-wemos-d1-r32.examples/advanced/psa_crypto:Run test
INFO:esp32-wemos-d1-r32.examples/advanced/psa_crypto:Run test.flash
WARNING:esp32-wemos-d1-r32.examples/advanced/psa_crypto:make RIOT_CI_BUILD=1 CC_NOCOLOR=1 --no-print-directory -C ./examples/advanced/psa_crypto flash-only
 You are going to use the PSA Crypto module, which is only partly implemented and not yet thouroughly tested.\n Please do not use this module in production, as it may introduce security issues!
/home/cbuec/RIOTstuff/riot-upstream/RIOT/dist/tools/esptools/venv/bin/esptool --chip esp32 --port /dev/ttyUSB0 --baud 460800 --before default-reset write-flash -z --flash-mode dout --flash-freq 40m --flash-size detect 0x1000 /home/cbuec/RIOTstuff/riot-upstream/RIOT/examples/advanced/psa_crypto/bin/esp32-wemos-d1-r32/esp_bootloader/bootloader.bin 0x8000 /home/cbuec/RIOTstuff/riot-upstream/RIOT/examples/advanced/psa_crypto/bin/esp32-wemos-d1-r32/partitions.bin 0x10000 /home/cbuec/RIOTstuff/riot-upstream/RIOT/examples/advanced/psa_crypto/bin/esp32-wemos-d1-r32/example_psa_crypto.elf.bin

 Usage: esptool write-flash [OPTIONS] <address> <filename>...

 Try 'esptool write-flash -h' for help
╭─ Error ──────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ Invalid value for '<address> <filename>...': [Errno 2] No such file or directory:                                    │
│ '/home/cbuec/RIOTstuff/riot-upstream/RIOT/examples/advanced/psa_crypto/bin/esp32-wemos-d1-r32/esp_bootloader/bootloa │
│ der.bin'                                                                                                             │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯

esptool v5.0.0
make: *** [/home/cbuec/RIOTstuff/riot-upstream/RIOT/examples/advanced/psa_crypto/../../../Makefile.include:845: flash-only] Error 2

Return value: 2

ERROR:esp32-wemos-d1-r32.examples/advanced/psa_crypto:Error during test.flash, writing to results/esp32-wemos-d1-r32/examples/advanced/psa_crypto/test.flash.failed
ERROR:esp32-wemos-d1-r32.examples/advanced/psa_crypto:Failed during: test.flash
```

Or you do it manually:
```
BOARD=esp32-wemos-d1-r32 make -C examples/advanced/psa_crypto all -j
BOARD=esp32-wemos-d1-r32 make -C examples/advanced/psa_crypto clean-intermediates
BOARD=esp32-wemos-d1-r32 make -C examples/advanced/psa_crypto flash-only
```


#### Expected results

It should flash.

#### Versions
Not relevant.

### Declaration of AI-Tools / LLMs usage:

AI-Tools / LLMs that were used are:
- none

--- a/Makefile.include
+++ b/Makefile.include
@@ -814,9 +814,13 @@ ifndef MAKE_RESTARTS
 endif
 
 # Remove intermediates, but keep the .elf, .hex and .map etc.
+# Special care has to be taken for ESP32 targets, the bootloader directory has
+# to be kept!
 clean-intermediates:
 	-@$(foreach dir,$(PKG_PATHS),"$(MAKE)" -C $(dir) distclean $(NEWLINE))
-	-@rm -rf $(BINDIR)/*.a $(BINDIR)/*/
+	-@rm -rf $(BINDIR)/*.a
+	-@find $(BINDIR)/ -mindepth 1 \
+	    -not -path "*/esp\_bootloader" -type d -exec rm -rf {} +;
 
 clean-pkg:
 	-@$(foreach dir,$(PKG_PATHS),"$(MAKE)" -C $(dir) distclean $(NEWLINE))

